### PR TITLE
Potential fix for watcher failing to collect events after about an hour

### PR DESCRIPTION
**Pull request by peter-moffatt**
This is a potential solution for issue https://github.com/splunk/fluent-plugin-kubernetes-objects/issues/9 which I am also experiencing. I have tested it in a Kubernetes cluster and I was able to successfully watch events for over 2 hours.

The fix is based on advice in https://github.com/abonas/kubeclient/issues/273.

This change breaks two of the unit tests which will need to be resolved before this is merged. I've never done any Ruby before so it'll take me a bit of time to figure out how to fix them. I will therefore delay figuring out how to fix the tests until I've got feedback that this is the right approach to fixing the problem.

### CODE REVIEW

**Comment by dtregonning** [comment]
Thanks @peter-moffatt - thanks for providing the link to the discussion and a potential fix. Its something that we've heard quite a few times. We have done some testing and this looks promising.  Will update soon with results.

**Comment by peter-moffatt** [comment]
Thanks for the update. I've had it watching events continuously for the last 4 days with no problems noticed so far.

File: lib/fluent/plugin/in_kubernetes_objects.rb [hunk 1]
```diff
@@ -176,9 +176,7 @@ def start_watchers
 	resource_name = o.delete(:resource_name)
 	version = @storage.get(resource_name)
 	o[:resource_version] = version if version
-	@client.public_send("watch_#{resource_name}", o).tap { |watcher|
-	  create_watcher_thread resource_name, watcher
-	}
+        create_watcher_thread resource_name, o
```
Comment (max-rocket-internet) [inline lib/fluent/plugin/in_kubernetes_objects.rb]: Is this extra space of indentation intentional?

Comment (peter-moffatt) [reply lib/fluent/plugin/in_kubernetes_objects.rb]: No it is not. I've fixed it.

**Comment by dtregonning** [comment]
@peter-moffatt - we are doing an update release to this project now with the plan to merge changes in after. Just a heads up the new project release wont have the fix.

**Comment by max-rocket-internet** [comment]
@dtregonning any update on this?

**Comment by mohitevishal** [comment]
Any update? Are you folks going to pull this patch? @dtregonning ?

**Comment by dtregonning** [comment]
Hey @mohitevishal, @max-rocket-internet  - yup. Team talked about it today. We will include this in a maintenance release and bump.

**Comment by max-rocket-internet** [comment]
@peter-moffatt could you rebase? And then perhaps we can merge?

**Comment by peter-moffatt** [comment]
> @peter-moffatt could you rebase? And then perhaps we can merge?

Done.

**Comment by max-rocket-internet** [comment]
Another conflict @peter-moffatt 😞

@peter-moffatt any update?

**Comment by dtregonning** [comment]
@max-rocket-internet @peter-moffatt - thank you so much for your efforts on this. We modified your original submitted fix and had it merged into develop to get it running in our testing environments. Im going to bump up the version of Objects to 1.1.1 with the fix included and release. Ill leave this open for now in case we hit any other issues.
https://github.com/splunk/fluent-plugin-kubernetes-objects/pull/21

**Comment by max-rocket-internet** [comment]
Thanks!

**Comment by peter-moffatt** [comment]
Thanks @dtregonning.

**Comment by max-rocket-internet** [comment]
How do we get this version from rubygems?

**Comment by dtregonning** [comment]
Fixed in #27

**Pull request closed without merging**